remote.go

```diff
@@ -0,0 +1,183 @@
+package blockstore
+
+import (
+	"context"
+	"io"
+
+	pb "github.com/RTradeLtd/TxPB/v3/go"
+	blocks "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+	"go.uber.org/zap"
+)
+
+var (
+	_ Blockstore = (*RemoteBlockstore)(nil)
+)
+
+// RemoteBlockstore storing blocks in remote locations
+// without running the blockstore locally. If behind latency
+// sensitive networks, it is advisable to wrap this in a cached blockstore
+type RemoteBlockstore struct {
+	ctx     context.Context
+	xclient pb.NodeAPIClient
+	logger  *zap.Logger
+}
+
+// NewRemoteBlockstore returns a new remote blockstore
+func NewRemoteBlockstore(ctx context.Context, logger *zap.Logger, xclient pb.NodeAPIClient) *RemoteBlockstore {
+	return &RemoteBlockstore{ctx, xclient, logger.Named("remote.blockstore")}
+}
+
+// DeleteBlock is used to delete the block from the remote blockstore.
+// This must be forced, as TemporalX does not allow "unforced" deletes
+func (rbs *RemoteBlockstore) DeleteBlock(gocid cid.Cid) error {
+	_, err := rbs.xclient.Blockstore(rbs.ctx, &pb.BlockstoreRequest{
+		RequestType: pb.BSREQTYPE_BS_DELETE,
+		ReqOpts:     []pb.BSREQOPTS{pb.BSREQOPTS_BS_FORCE},
+		Cids:        []string{gocid.String()},
+	})
+	if err != nil {
+		rbs.logger.Error("delete block failed", zap.Error(err), zap.String("cid", gocid.String()))
+	}
+	return err
+}
+
+// Has is used to check whether or not we have the block
+func (rbs *RemoteBlockstore) Has(gocid cid.Cid) (bool, error) {
+	resp, err := rbs.xclient.Blockstore(rbs.ctx, &pb.BlockstoreRequest{
+		RequestType: pb.BSREQTYPE_BS_HAS,
+		Cids:        []string{gocid.String()},
+	})
+	if err != nil {
+		rbs.logger.Error("has check failed", zap.Error(err), zap.String("cid", gocid.String()))
+		return false, err
+	}
+	if len(resp.GetBlocks()) <= 0 {
+		return false, ErrNotFound
+	}
```

Comment:
This part is duplicated 3 times with only two differences. Consider creating a helper fuction.

Reply:
good catch